Content.Server/Pinpointer/NavMapSystem.cs

```diff
@@ -59,6 +59,8 @@ private void OnStationInit(StationGridAddedEvent ev)
 
     private void OnNavMapBeaconStartup(EntityUid uid, NavMapBeaconComponent component, ComponentStartup args)
     {
+        component.Text ??= string.Empty;
+        component.Text = Loc.GetString(component.Text);
```

Comment:
If I do it only at mapinit, then beacons spawned or ordered during the round will not be localized. 
Should I just keep Text (holding the prototype-derived-locale-identifier) separate from a new string variable that holds what text players entered, if any? (This will cause a change in behavior, that an "empty" text input for the beacon will make it fall back to the prototype's default text, rather than the entity name

Reply:
mapinit is raised on entity spawn. It's essentially componentstartup but raised only once for the entire entity lifespan.

Reply:
Thanks Emo